htdocs/commande/class/commande.class.php

```diff
@@ -1706,7 +1706,6 @@ public function addline($desc, $pu_ht, $qty, $txtva, $txlocaltax1 = 0, $txlocalt
 
 				if ($result > 0) {
 					$this->db->commit();
-					$this->lines[] = $this->line;
```

Comment:
I you remove this line, if order has 2 lines before adding the line, the order will stil have 2 lines after when it should have 3 lines.
Adding a resorting of lines would be better
For example by using  $this->lines = dol_sort_array($this->lines, 'rang')